packages/widget/src/components/common/buttons/connect-wallet.ts

```diff
@@ -62,49 +61,124 @@ export class ConnectWalletButton extends BaseComponent {
     return !!this.wallets.evmWallet || !!this.wallets.substrateWallet;
   }
 
-  render(): HTMLTemplateResult {
-    const evmWallet = this.wallets.evmWallet;
+  private handleSubstrateAccountSelected = (
+    option: DropdownOption<Account>
+  ): void => {
+    this.dispatchEvent(
+      new WalletUpdateEvent({
+        substrateWallet: {
+          signer: this.wallets.substrateWallet!.signer,
+          signerAddress: option.value.address,
+          accounts: this.wallets.substrateWallet?.accounts ?? []
+        }
+      })
+    );
+  };
+
+  private renderDisconnectSubstrateButton(): HTMLTemplateResult | undefined {
+    return html` <div
+      class="dropdownOption substrateDisconnectButton"
+      part="substrateDisconnectButton"
+      @click=${this.onDisconnectClicked}
+    >
+      Disconnect
+    </div>`;
+  }
+
+  private renderSubstrateAccount(): HTMLTemplateResult {
     const substrateWallet = this.wallets.substrateWallet;
-    //TODO: this is wrong we need to enable user to select account
+    if (!substrateWallet) return html``;
+
+    function renderCustomOptionContent({
```

Comment:
is it possible to move this two functions `renderCustomOptionContent` and `normalizeOptionsData` from the lexical scope of this method? To me it looks strange to have it here embedded in this method

Reply:
Yeah, it looks unusual, but it also tells that it belongs only to "renderSubstrateAccount" method. Nevertheless, I can move it 👍